packages/plugin-hardhat/src/type-extensions.ts

```diff
@@ -5,5 +5,22 @@ import type { HardhatUpgrades } from '.';
 declare module 'hardhat/types/runtime' {
   export interface HardhatRuntimeEnvironment {
     upgrades: HardhatUpgrades;
+    platform: HardhatUpgrades;
+  }
+}
+
+export interface HardhatPlatformConfig {
+  apiKey: string;
+  apiSecret: string;
+  deploy?: boolean;
```

Comment:
This option name is a little confusing I think. Perhaps we should do `useDeploy` or `preferDeploy`.